2017-04-28 Jingyue Shen <brainshen@g.ucla.edu>

    * lib/timespec.h: use a conditional statement to prevent overflow 
      		      when comparing modification time of two files.
		      This change eliminates the overflow problem caused
		      by ancient files.

--- a/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ b/lib/timespec.h	2017-04-28 16:45:39.124869000 -0700
@@ -46,7 +46,9 @@
 timespec_cmp (struct timespec a, struct timespec b)
 {
   int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if((a.tv_sec > 0 && b.tv_sec < 0 && diff < 0) || (a.tv_sec < 0 && b.tv_sec > 0 && diff > 0))
+    diff = a.tv_sec;  
+ return diff ? diff : a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
